server/api/utils/builder.py

```diff
@@ -160,31 +160,22 @@ def make_kaniko_pod(
     extra_args="",
     extra_labels=None,
     project_secrets=None,
+    project_default_fucntion_node_selector=None,
 ):
     extra_runtime_spec = {}
     if not registry:
         # if registry was not given, infer it from the image destination
         registry = dest.partition("/")[0]
 
     # set kaniko's spec attributes from the runtime spec
-    for attribute in get_kaniko_spec_attributes_from_runtime():
-        attr_value = getattr(runtime_spec, attribute, None)
-        if attribute == "service_account":
-            from server.api.api.utils import resolve_project_default_service_account
-
-            (
-                allowed_service_accounts,
-                default_service_account,
-            ) = resolve_project_default_service_account(project)
-            if attr_value:
-                runtime_spec.validate_service_account(allowed_service_accounts)
-            else:
-                attr_value = default_service_account
-
-        if not attr_value:
-            continue
-
-        extra_runtime_spec[attribute] = attr_value
+    for attribute, handler in get_kaniko_spec_attributes_from_runtime(
+        project,
+        runtime_spec,
+        project_default_fucntion_node_selector,
+    ).items():
+        attr_value = handler(getattr(runtime_spec, attribute, None))
```

Comment:
you call the handler 2 lines below
```suggestion
        attr_value = getattr(runtime_spec, attribute, None)
```

Reply:
why? I need to call the handler here